### PR TITLE
feature/file upload

### PR DESCRIPTION
**Pull request by DilnazTolubaiKyzy**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -46,59 +51,32 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
-		<dependency>
```
Comment (i-Coderr) [inline pom.xml:49]: Add those dependencies again pls, why remove them?

File: pom.xml [hunk 2]
```diff
@@ -46,59 +51,32 @@
...
-		</dependency>
-		<dependency>
-			<groupId>org.springdoc</groupId>
-			<artifactId>springdoc-openapi-ui</artifactId>
-			<version>1.6.6</version>
-		</dependency>
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-security</artifactId>
+			<artifactId>spring-boot-starter-web</artifactId>
```
Comment (i-Coderr) [inline pom.xml:56]: Why changed to web? There is already web dependency on line 30

File: src/main/java/us/peaksoft/gadgetarium/s3/S3Config.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class S3Config {
+    @Value("${cloud.aws.credentials.access-key}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/s3/S3Config.java:14]: next line

File: src/main/java/us/peaksoft/gadgetarium/s3/S3Config.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    private String accessKey;
+
+    @Value("${cloud.aws.credentials.secret-key}")
+    private String accessSecret;
+    @Value("${cloud.aws.region.static}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/s3/S3Config.java:19]: next line

File: src/main/java/us/peaksoft/gadgetarium/s3/S3Service.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+@Service
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/s3/S3Service.java:17]: next line

File: src/main/java/us/peaksoft/gadgetarium/s3/S3Service.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+@Slf4j
+@RequiredArgsConstructor
+public class S3Service{
+    @Value("${application.bucket.name}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/s3/S3Service.java:21]: next line

File: src/main/java/us/peaksoft/gadgetarium/s3/S3StorageController.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package us.peaksoft.gadgetarium.s3;
+import lombok.RequiredArgsConstructor;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/s3/S3StorageController.java:9]: remove one line

**Review by i-Coderr — COMMENTED** [review]
Please add in application.properties file s3 configs

File: pom.xml (hunk 1 above)
Comment (DilnazTolubaiKyzy) [reply pom.xml:49]: i didn't delete anything

**Review by i-Coderr — APPROVED** [review]
Alright

**Pull request merged**